user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementMain.java

```diff
@@ -283,50 +287,76 @@ private Dialog setupDialog(MeasurementMetadataUploadDialog dialog) {
     dialog.addCancelListener(cancelEvent -> cancelEvent.getSource().close());
     dialog.addConfirmListener(confirmEvent ->
     {
-      triggerMeasurementRegistration(confirmEvent.uploads(),
-          confirmEvent.getSource());
-      setMeasurementInformation();
+      triggerMeasurementRegistration(confirmEvent);
     });
     return dialog;
   }
 
   private void triggerMeasurementRegistration(
-      List<MeasurementMetadataUpload<MeasurementMetadata>> measurementMetadataUploads,
-      MeasurementMetadataUploadDialog measurementMetadataUploadDialog) {
-    String process =
-        measurementMetadataUploadDialog.getMode() == MODE.EDIT ? "update" : "registration";
-    for (var upload : measurementMetadataUploads) {
+      ConfirmEvent event) {
+    var uploads = new ArrayList<>(event.uploads());
+    var dialog = event.getSource();
+    var mode = dialog.getMode();
+    UI ui = event.getSource().getUI().orElseThrow();
+    for (var upload : uploads) {
       var measurementData = upload.measurementMetadata();
-      measurementMetadataUploadDialog.taskInProgress(
-          "%s of %s measurements ...".formatted(StringUtils.capitalize(process),
-              measurementData.size()),
-          "This might take a minute");
+
+     // UI ui = UI.getCurrent();
       //Necessary so the dialog window switches to show the upload progress
-      UI.getCurrent().push();
+      ui.push();
       CompletableFuture<List<Result<MeasurementId, MeasurementService.ErrorCode>>> completableFuture;
-      if (measurementMetadataUploadDialog.getMode().equals(MODE.EDIT)) {
-        completableFuture = measurementService.updateAll(upload.measurementMetadata(),
+      Toast pendingToast;
+
+      // we can close the dialog, everything that comes now is executed concurrently and there is
+      // no need the user is blocked
+      event.getSource().close();
+
+      if (mode.equals(MODE.EDIT)) {
+        completableFuture = measurementService.updateAll(measurementData,
             context.projectId().orElseThrow());
+        pendingToast = messageFactory.pendingTaskToast("task.in-progress", new Object[]{
+                "Update of #%d measurements".formatted(measurementData.size())},
```

Comment:
This can be confusing in the pooling context, since it will show the number of rows which are updated but not the number of unique measurements.

Reply:
What do you suggest instead?

Reply:
so straight forward, love it. will do